web/src/lottie-player.ts

```diff
@@ -0,0 +1,577 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { html, PropertyValueMap, LitElement, type TemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+// @ts-ignore: WASM Glue code doesn't have type & Only available on build progress
+import Module from '../dist/thorvg-wasm';
+import { THORVG_VERSION } from './version';
+
+let _tvg: any;
+let _module: any;
+(async () => {  
+  _module = await Module();
+  _tvg = new _module.TvgLottieAnimation();
+})();
+
+// Define library version
+export interface LibraryVersion {
+  THORVG_VERSION: string
+}
+
+// Define file type which can be exported
+export enum ExportableType {
+  GIF = 'gif',
+}
+
+// Define valid player states
+export enum PlayerState {
+ Destroyed = "destroyed", // Player is destroyed by `destroy()` method
+ Error = "error", // An error occurred
+ Loading = "loading", // Player is loading
+ Paused = "paused", // Player is paused
+ Playing = "playing", // Player is playing
+ Stopped = "stopped",  // Player is stopped
+ Frozen = "frozen", // Player is paused due to player being invisible
+}
+
+// Define play modes
+export enum PlayMode {
+  Bounce = "bounce",
+  Normal = "normal",
+}
+
+// Define player events
+export enum PlayerEvent {
+  Complete = "complete",
+  Destroyed = "destroyed",
+  Error = "error",
+  Frame = "frame",
+  Freeze = "freeze",
+  Load = "load",
+  Loop = "loop",
+  Pause = "pause",
+  Play = "play",
+  Ready = "ready",
+  Stop = "stop",
+  }
+
+/**
+ * Parse a resource into a bytes
+ */
+const _parseSrc = async (src: string | object): Promise<Uint8Array> => {
+  let data = src;
+  if (typeof data === "object") {
+    data = JSON.stringify(data);
+  } else if (typeof data === "string") {
+    try {
+      data = JSON.parse(data);
+    } catch (err) {
+      const json = await _fromURL(data as string);
+      data = JSON.stringify(json);
+    }
+  }
+
+  const encoder = new TextEncoder();
+  return encoder.encode(data as string);
+}
+
+const _fromURL = async (url: string): Promise<Record<string, any>> => {
+  if (typeof url !== "string") {
+    throw new Error(`The url value must be a string`);
+  }
+
+  let json;
+
+  try {
+    // Try construct an absolute URL from the src URL
+    const srcUrl: URL = new URL(url);
+
+    // Fetch the JSON file from the URL
+    const result: any = await fetch(srcUrl.toString());
+
+    json = await result.json();
+  } catch (err) {
+    throw new Error(
+      `An error occurred while trying to load the Lottie file from URL`
+    );
+  }
+
+  return json;
+}
+
+const _wait = (timeToDelay: number) => {
```

Comment:
you don't need the underscore prefix for helper functions that are outside of the class. You should use it only for private variables or methods inside a class or for unused variables.

```suggestion
const wait = (timeToDelay: number) => {
```